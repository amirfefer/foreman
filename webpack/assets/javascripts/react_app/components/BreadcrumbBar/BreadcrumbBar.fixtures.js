export const breadcrumbItems = {
  items: [
    {
      caption: 'root',
      url: '/some-url',
    },
    {
      caption: 'child',
    },
  ],
};

export const breadcrumbTitleItems = {
  items: [
    {
      caption: 'title',
    },
  ],
};

export const resource = {
  resourceUrl: 'some/url',
  nameField: 'name',
  switcherItemUrl: 'some/url/:id',
};

export const resourceList = [
  { name: 'Host 1', url: '#' },
  { name: 'Host 2', url: '#' },
  { name: 'Host 3 with a very long name', url: '#' },
  { name: 'Host 4', url: undefined, onClick: jest.fn() },
  { name: 'Host 5', url: '#', onClick: undefined },
];

export const serverResourceListResponse = {
  data: {
    page: 1,
    total: 3,
    per_page: 2,
    results: [
      { id: '1', name: 'name-1' },
      { id: '2', name: 'name-2' },
      { id: '3', name: 'name-3' },
    ],
  },
};

export const breadcrumbSwitcherLoading = {
  loading: true,
  resources: [],
};

export const breadcrumbSwitcherLoaded = {
  loading: false,
  resources: resourceList,
};

export const breadcrumbSwitcherLoadedWithPagination = {
  ...breadcrumbSwitcherLoaded,
  currentPage: 2,
  totalPages: 3,
};

export const breadcrumbBar = {
  data: {
    resource,
    breadcrumbItems: breadcrumbItems.items,
    isSwitchable: false,
  },
};

export const breadcrumbBarSwithcable = {
  data: {
    resource,
    breadcrumbItems: breadcrumbItems.items,
    isSwitchable: true,
  },
};
