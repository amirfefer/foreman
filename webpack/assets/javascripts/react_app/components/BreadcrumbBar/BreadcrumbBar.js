import React from 'react';
import Breadcrumbs from './components/Breadcrumb';
import Switcher from './components/BreadcrumbSwitcher';
import API from '../../API';
import Helpers from '../../common/helpers';

class BreadcrumbBar extends React.Component {
  constructor(props) {
    super(props);
    this.state = { menu: [], resurceItems: [], switchable: true };
  }
  componentDidMount() {
    const { data: { reosurceUrl, name, menu, resource, action } } = this.props;
    API.get(reosurceUrl).then(({ data }) => {
      const newData = data.results.map(res => ({ name: res[name], url: Helpers.urlBuilder(resource, action, res.id) }));
      this.setState({ resurceItems: newData });
    });
    this.setState({ menu });
  }

  render() {
    const { menu, resurceItems } = this.state;
    return (
      <div>
        <Breadcrumbs data={{ menu }} >
        {this.state.switchable &&
        <Switcher resources={resurceItems} />
        }
        </Breadcrumbs>
        <hr className="breadcrumb-line" />
      </div>
    );
  }
}

export default BreadcrumbBar;
