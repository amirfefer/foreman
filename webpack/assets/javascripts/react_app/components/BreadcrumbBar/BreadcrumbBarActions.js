import API from '../../API';

import {
  BREADCRUMB_BAR_TOGGLE_SWITCHER,
  BREADCRUMB_BAR_CLOSE_SWITCHER,
  BREADCRUMB_BAR_RESOURCES_REQUEST,
  BREADCRUMB_BAR_RESOURCES_SUCCESS,
  BREADCRUMB_BAR_RESOURCES_FAILURE,
} from './BreadcrumbBarConstants';

export const toggleSwitcher = () => ({
  type: BREADCRUMB_BAR_TOGGLE_SWITCHER,
});

export const closeSwitcher = () => ({
  type: BREADCRUMB_BAR_CLOSE_SWITCHER,
});

export const loadSwitcherResourcesByResource = (resource, options = {}) => (dispatch) => {
  const {
    reosurceUrl, nameField, switcherItemUrl,
  } = resource;
  const { page = 1 } = options;

  const beforeRequest = () =>
    dispatch({
      type: BREADCRUMB_BAR_RESOURCES_REQUEST,
      payload: { resource, options },
    });

  const onRequestSuccess = response =>
    dispatch({ type: BREADCRUMB_BAR_RESOURCES_SUCCESS, payload: formatResults(response) });

  const onRequestFail = error =>
    dispatch({ type: BREADCRUMB_BAR_RESOURCES_FAILURE, payload: error });

  const formatResults = ({ data }) => {
    const switcherItems = Object.values(data.results).map((x, i) =>
      (data.results instanceof Array ?
        { name: x[nameField], url: switcherItemUrl.replace(':id', x.id) } :
        { name: x[i][nameField], url: switcherItemUrl.replace(':id', x[i].id) }));
    return {
      items: switcherItems,
      page: data.page,
      pages: Number(data.total) / Number(data.per_page),
    };
  };
  beforeRequest();

  return API.get(reosurceUrl, {}, { page }).then(onRequestSuccess, onRequestFail);
};
