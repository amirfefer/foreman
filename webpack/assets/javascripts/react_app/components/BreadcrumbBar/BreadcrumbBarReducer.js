import Immutable from 'seamless-immutable';

import {
  BREADCRUMB_BAR_TOGGLE_SWITCHER,
  BREADCRUMB_BAR_RESOURCES_REQUEST,
  BREADCRUMB_BAR_RESOURCES_SUCCESS,
  BREADCRUMB_BAR_RESOURCES_FAILURE,
} from './BreadcrumbBarConstants';

const initialState = Immutable({
  resourceSwitcherItems: [],
  isLoading: false,
  isSwitcherOpen: false,
  requestError: null,
});

export default (state = initialState, action) => {
  const { payload } = action;

  switch (action.type) {
    case BREADCRUMB_BAR_RESOURCES_REQUEST:
      return state
        .set('resourceSwitcherItems', [])
        .set('requestError', null)
        .set('isLoading', true);

    case BREADCRUMB_BAR_RESOURCES_SUCCESS:
      return state
        .set('resourceSwitcherItems', payload)
        .set('requestError', null)
        .set('isLoading', false);

    case BREADCRUMB_BAR_RESOURCES_FAILURE:
      return state
        .set('resourceSwitcherItems', [])
        .set('requestError', payload)
        .set('isLoading', false);

    case BREADCRUMB_BAR_TOGGLE_SWITCHER:
      return state.set('isSwitcherOpen', !state.isSwitcherOpen);

    default:
      return state;
  }
};
