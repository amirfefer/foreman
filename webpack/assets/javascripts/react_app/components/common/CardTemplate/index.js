import React from 'react';
import CardItem from './CardItem';

import { addGlobalFill } from '../../common/Fill/GlobalFill';

const CardTemplate = ({ content, header, slotID, cardID }) => {
  addGlobalFill(
    slotID,
    cardID,
    <CardItem content={content} header={header} />,
    200
  );
};

export default CardTemplate;
