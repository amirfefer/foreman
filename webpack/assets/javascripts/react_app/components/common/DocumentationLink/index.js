import React from 'react';
import { MenuItem } from 'patternfly-react';
import PropTypes from 'prop-types';
import Icon from '../Icon';

const DocumentationLink = ({ href, id }) => {
  const handleClick = e => {
    e.preventDefault();
    window.open(href, '_blank');
  };

  return (
    <MenuItem key="documentationUrl" id={id} href={href} onClick={handleClick}>
      <Icon type="question-sign" className="icon-black" />
      {` ${__('Documentation')}`}
    </MenuItem>
  );
};

export default DocumentationLink;

DocumentationLink.propTypes = {
  href: PropTypes.string.isRequired,
  id: PropTypes.string,
};

DocumentationLink.defaultProps = {
  id: 'documentationLink',
};
