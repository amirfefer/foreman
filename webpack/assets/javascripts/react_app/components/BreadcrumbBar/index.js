import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';

import * as actions from './BreadcrumbBarActions';
import reducer from './BreadcrumbBarReducer';

import BreadcrumbBar from './BreadcrumbBar';

// map state to props
const mapStateToProps = ({ breadcrumbBar }) => ({
  resourceSwitcherItems: breadcrumbBar.resourceSwitcherItems,
  isSwitcherOpen: breadcrumbBar.isSwitcherOpen,
  isLoadingResources: breadcrumbBar.isLoadingResources,
  requestError: breadcrumbBar.requestError,
});

// map action dispatchers to props
const mapDispatchToProps = dispatch => bindActionCreators(actions, dispatch);

// export reducers
export const reducers = { breadcrumbBar: reducer };

// export connected component
export default connect(mapStateToProps, mapDispatchToProps)(BreadcrumbBar);
