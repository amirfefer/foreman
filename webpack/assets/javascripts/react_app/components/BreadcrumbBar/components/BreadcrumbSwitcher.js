import React from 'react';
import ReactDOM from 'react-dom';
import PropTypes from 'prop-types';
import { Overlay } from 'react-bootstrap';

import { noop } from '../../../common/helpers';

import BreadcrumbSwitcherPopover from './BreadcrumbSwitcherPopover';
import BreadcrumbSwitcherToggler from './BreadcrumbSwitcherToggler';

class BreadcrumbSwitcher extends React.Component {
  render() {
    const {
      open,
      currentPage,
      totalPages,
      isLoadingResources,
      hasError,
      resources,
      onTogglerClick,
      onOverlayHide,
      onOverlayEnter,
      onNextPageClick,
      onPrevPageClick,
    } = this.props;

    return (
      <div className="breadcrumb-switcher" style={{ position: 'relative' }}>
        <BreadcrumbSwitcherToggler
          onClick={() => onTogglerClick()}
          ref={(ref) => {
            this.togglerRef = ref;
          }}
        />

        <Overlay
          rootClose
          show={open}
          container={this}
          placement="bottom"
          onHide={() => onOverlayHide()}
          onEnter={() => onOverlayEnter()}
          target={() => ReactDOM.findDOMNode(this.togglerRef)}
        >
          <BreadcrumbSwitcherPopover
            id="breadcrumb-switcher-popover"
            loading={isLoadingResources}
            hasError={hasError}
            resources={resources}
            onNextPageClick={() => onNextPageClick()}
            onPrevPageClick={() => onPrevPageClick()}
            currentPage={currentPage}
            totalPages={totalPages}
          />
        </Overlay>
      </div>
    );
  }
}

BreadcrumbSwitcher.propTypes = {
  open: PropTypes.bool,
  currentPage: PropTypes.number,
  totalPages: PropTypes.number,
  isLoadingResources: PropTypes.bool,
  hasError: PropTypes.bool,
  resources: BreadcrumbSwitcherPopover.propTypes.resources,
  onTogglerClick: PropTypes.func,
  onOverlayHide: PropTypes.func,
  onOverlayEnter: PropTypes.func,
  onPrevPageClick: PropTypes.func,
  onNextPageClick: PropTypes.func,
};

BreadcrumbSwitcher.defaultProps = {
  open: false,
  currentPage: 1,
  totalPages: 1,
  isLoadingResources: false,
  hasError: false,
  resources: [],
  onTogglerClick: noop,
  onOverlayHide: noop,
  onOverlayEnter: noop,
  onPrevPageClick: noop,
  onNextPageClick: noop,
};

export default BreadcrumbSwitcher;
