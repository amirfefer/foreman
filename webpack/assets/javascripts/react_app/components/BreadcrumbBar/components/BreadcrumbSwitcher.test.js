import { testComponentSnapshotsWithFixtures } from '../../../common/testHelpers';

import BreadcrumbSwitcher from './BreadcrumbSwitcher';
import { breadcrumbSwitcherLoading, breadcrumbSwitcherLoaded } from '../BreadcrumbBar.fixtures';

const fixtures = {
  'render loading state': breadcrumbSwitcherLoading,
  'render resources list': breadcrumbSwitcherLoaded,
};

describe('BreadcrumbSwitcher', () => testComponentSnapshotsWithFixtures(BreadcrumbSwitcher, fixtures));
