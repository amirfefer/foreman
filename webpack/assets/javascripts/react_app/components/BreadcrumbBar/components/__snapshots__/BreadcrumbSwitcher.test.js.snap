// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BreadcrumbSwitcher rendering render closed 1`] = `
<div
  className="breadcrumb-switcher"
  style={
    Object {
      "position": "relative",
    }
  }
>
  <BreadcrumbSwitcherToggler
    onClick={[Function]}
  />
  <Overlay
    animation={[Function]}
    container={
      BreadcrumbSwitcher {
        "context": Object {},
        "props": Object {
          "currentPage": 1,
          "hasError": false,
          "isLoadingResources": false,
          "onNextPageClick": [Function],
          "onOverlayEnter": [Function],
          "onOverlayHide": [Function],
          "onPrevPageClick": [Function],
          "onTogglerClick": [Function],
          "open": false,
          "resources": Array [],
          "totalPages": 1,
        },
        "refs": Object {},
        "state": null,
        "updater": Updater {
          "_callbacks": Array [],
          "_renderer": ReactShallowRenderer {
            "_context": Object {},
            "_element": <BreadcrumbSwitcher
              currentPage={1}
              hasError={false}
              isLoadingResources={false}
              onNextPageClick={[Function]}
              onOverlayEnter={[Function]}
              onOverlayHide={[Function]}
              onPrevPageClick={[Function]}
              onTogglerClick={[Function]}
              open={false}
              resources={Array []}
              totalPages={1}
      />,
            "_forcedUpdate": false,
            "_instance": [Circular],
            "_newState": null,
            "_rendered": <div
              className="breadcrumb-switcher"
              style={
                      Object {
                              "position": "relative",
                            }
              }
      >
              <BreadcrumbSwitcherToggler
                      onClick={[Function]}
              />
              <Overlay
                      animation={[Function]}
                      container={[Circular]}
                      onEnter={[Function]}
                      onHide={[Function]}
                      placement="bottom"
                      rootClose={true}
                      show={false}
                      target={[Function]}
              >
                      <BreadcrumbSwitcherPopover
                              currentPage={1}
                              hasError={false}
                              id="breadcrumb-switcher-popover"
                              loading={false}
                              onNextPageClick={[Function]}
                              onPrevPageClick={[Function]}
                              resources={Array []}
                              totalPages={1}
                      />
              </Overlay>
      </div>,
            "_rendering": false,
            "_updater": [Circular],
          },
        },
      }
    }
    onEnter={[Function]}
    onHide={[Function]}
    placement="bottom"
    rootClose={true}
    show={false}
    target={[Function]}
  >
    <BreadcrumbSwitcherPopover
      currentPage={1}
      hasError={false}
      id="breadcrumb-switcher-popover"
      loading={false}
      onNextPageClick={[Function]}
      onPrevPageClick={[Function]}
      resources={Array []}
      totalPages={1}
    />
  </Overlay>
</div>
`;

exports[`BreadcrumbSwitcher rendering render loading state 1`] = `
<div
  className="breadcrumb-switcher"
  style={
    Object {
      "position": "relative",
    }
  }
>
  <BreadcrumbSwitcherToggler
    onClick={[Function]}
  />
  <Overlay
    animation={[Function]}
    container={
      BreadcrumbSwitcher {
        "context": Object {},
        "props": Object {
          "currentPage": 1,
          "hasError": false,
          "isLoadingResources": false,
          "loading": true,
          "onNextPageClick": [Function],
          "onOverlayEnter": [Function],
          "onOverlayHide": [Function],
          "onPrevPageClick": [Function],
          "onTogglerClick": [Function],
          "open": true,
          "resources": Array [],
          "totalPages": 1,
        },
        "refs": Object {},
        "state": null,
        "updater": Updater {
          "_callbacks": Array [],
          "_renderer": ReactShallowRenderer {
            "_context": Object {},
            "_element": <BreadcrumbSwitcher
              currentPage={1}
              hasError={false}
              isLoadingResources={false}
              loading={true}
              onNextPageClick={[Function]}
              onOverlayEnter={[Function]}
              onOverlayHide={[Function]}
              onPrevPageClick={[Function]}
              onTogglerClick={[Function]}
              open={true}
              resources={Array []}
              totalPages={1}
      />,
            "_forcedUpdate": false,
            "_instance": [Circular],
            "_newState": null,
            "_rendered": <div
              className="breadcrumb-switcher"
              style={
                      Object {
                              "position": "relative",
                            }
              }
      >
              <BreadcrumbSwitcherToggler
                      onClick={[Function]}
              />
              <Overlay
                      animation={[Function]}
                      container={[Circular]}
                      onEnter={[Function]}
                      onHide={[Function]}
                      placement="bottom"
                      rootClose={true}
                      show={true}
                      target={[Function]}
              >
                      <BreadcrumbSwitcherPopover
                              currentPage={1}
                              hasError={false}
                              id="breadcrumb-switcher-popover"
                              loading={false}
                              onNextPageClick={[Function]}
                              onPrevPageClick={[Function]}
                              resources={Array []}
                              totalPages={1}
                      />
              </Overlay>
      </div>,
            "_rendering": false,
            "_updater": [Circular],
          },
        },
      }
    }
    onEnter={[Function]}
    onHide={[Function]}
    placement="bottom"
    rootClose={true}
    show={true}
    target={[Function]}
  >
    <BreadcrumbSwitcherPopover
      currentPage={1}
      hasError={false}
      id="breadcrumb-switcher-popover"
      loading={false}
      onNextPageClick={[Function]}
      onPrevPageClick={[Function]}
      resources={Array []}
      totalPages={1}
    />
  </Overlay>
</div>
`;

exports[`BreadcrumbSwitcher rendering render resources list 1`] = `
<div
  className="breadcrumb-switcher"
  style={
    Object {
      "position": "relative",
    }
  }
>
  <BreadcrumbSwitcherToggler
    onClick={[Function]}
  />
  <Overlay
    animation={[Function]}
    container={
      BreadcrumbSwitcher {
        "context": Object {},
        "props": Object {
          "currentPage": 1,
          "hasError": false,
          "isLoadingResources": false,
          "loading": false,
          "onNextPageClick": [Function],
          "onOverlayEnter": [Function],
          "onOverlayHide": [Function],
          "onPrevPageClick": [Function],
          "onTogglerClick": [Function],
          "open": true,
          "resources": Array [
            Object {
              "name": "Host 1",
              "url": "#",
            },
            Object {
              "name": "Host 2",
              "url": "#",
            },
            Object {
              "name": "Host 3 with a very long name",
              "url": "#",
            },
            Object {
              "name": "Host 4",
              "onClick": [Function],
              "url": undefined,
            },
            Object {
              "name": "Host 5",
              "onClick": undefined,
              "url": "#",
            },
          ],
          "totalPages": 1,
        },
        "refs": Object {},
        "state": null,
        "updater": Updater {
          "_callbacks": Array [],
          "_renderer": ReactShallowRenderer {
            "_context": Object {},
            "_element": <BreadcrumbSwitcher
              currentPage={1}
              hasError={false}
              isLoadingResources={false}
              loading={false}
              onNextPageClick={[Function]}
              onOverlayEnter={[Function]}
              onOverlayHide={[Function]}
              onPrevPageClick={[Function]}
              onTogglerClick={[Function]}
              open={true}
              resources={
                      Array [
                              Object {
                                "name": "Host 1",
                                "url": "#",
                              },
                              Object {
                                "name": "Host 2",
                                "url": "#",
                              },
                              Object {
                                "name": "Host 3 with a very long name",
                                "url": "#",
                              },
                              Object {
                                "name": "Host 4",
                                "onClick": [Function],
                                "url": undefined,
                              },
                              Object {
                                "name": "Host 5",
                                "onClick": undefined,
                                "url": "#",
                              },
                            ]
              }
              totalPages={1}
      />,
            "_forcedUpdate": false,
            "_instance": [Circular],
            "_newState": null,
            "_rendered": <div
              className="breadcrumb-switcher"
              style={
                      Object {
                              "position": "relative",
                            }
              }
      >
              <BreadcrumbSwitcherToggler
                      onClick={[Function]}
              />
              <Overlay
                      animation={[Function]}
                      container={[Circular]}
                      onEnter={[Function]}
                      onHide={[Function]}
                      placement="bottom"
                      rootClose={true}
                      show={true}
                      target={[Function]}
              >
                      <BreadcrumbSwitcherPopover
                              currentPage={1}
                              hasError={false}
                              id="breadcrumb-switcher-popover"
                              loading={false}
                              onNextPageClick={[Function]}
                              onPrevPageClick={[Function]}
                              resources={
                                      Array [
                                              Object {
                                                "name": "Host 1",
                                                "url": "#",
                                              },
                                              Object {
                                                "name": "Host 2",
                                                "url": "#",
                                              },
                                              Object {
                                                "name": "Host 3 with a very long name",
                                                "url": "#",
                                              },
                                              Object {
                                                "name": "Host 4",
                                                "onClick": [Function],
                                                "url": undefined,
                                              },
                                              Object {
                                                "name": "Host 5",
                                                "onClick": undefined,
                                                "url": "#",
                                              },
                                            ]
                              }
                              totalPages={1}
                      />
              </Overlay>
      </div>,
            "_rendering": false,
            "_updater": [Circular],
          },
        },
      }
    }
    onEnter={[Function]}
    onHide={[Function]}
    placement="bottom"
    rootClose={true}
    show={true}
    target={[Function]}
  >
    <BreadcrumbSwitcherPopover
      currentPage={1}
      hasError={false}
      id="breadcrumb-switcher-popover"
      loading={false}
      onNextPageClick={[Function]}
      onPrevPageClick={[Function]}
      resources={
        Array [
          Object {
            "name": "Host 1",
            "url": "#",
          },
          Object {
            "name": "Host 2",
            "url": "#",
          },
          Object {
            "name": "Host 3 with a very long name",
            "url": "#",
          },
          Object {
            "name": "Host 4",
            "onClick": [Function],
            "url": undefined,
          },
          Object {
            "name": "Host 5",
            "onClick": undefined,
            "url": "#",
          },
        ]
      }
      totalPages={1}
    />
  </Overlay>
</div>
`;

exports[`BreadcrumbSwitcher rendering render resources list with pagination 1`] = `
<div
  className="breadcrumb-switcher"
  style={
    Object {
      "position": "relative",
    }
  }
>
  <BreadcrumbSwitcherToggler
    onClick={[Function]}
  />
  <Overlay
    animation={[Function]}
    container={
      BreadcrumbSwitcher {
        "context": Object {},
        "props": Object {
          "currentPage": 2,
          "hasError": false,
          "isLoadingResources": false,
          "loading": false,
          "onNextPageClick": [Function],
          "onOverlayEnter": [Function],
          "onOverlayHide": [Function],
          "onPrevPageClick": [Function],
          "onTogglerClick": [Function],
          "open": true,
          "resources": Array [
            Object {
              "name": "Host 1",
              "url": "#",
            },
            Object {
              "name": "Host 2",
              "url": "#",
            },
            Object {
              "name": "Host 3 with a very long name",
              "url": "#",
            },
            Object {
              "name": "Host 4",
              "onClick": [Function],
              "url": undefined,
            },
            Object {
              "name": "Host 5",
              "onClick": undefined,
              "url": "#",
            },
          ],
          "totalPages": 3,
        },
        "refs": Object {},
        "state": null,
        "updater": Updater {
          "_callbacks": Array [],
          "_renderer": ReactShallowRenderer {
            "_context": Object {},
            "_element": <BreadcrumbSwitcher
              currentPage={2}
              hasError={false}
              isLoadingResources={false}
              loading={false}
              onNextPageClick={[Function]}
              onOverlayEnter={[Function]}
              onOverlayHide={[Function]}
              onPrevPageClick={[Function]}
              onTogglerClick={[Function]}
              open={true}
              resources={
                      Array [
                              Object {
                                "name": "Host 1",
                                "url": "#",
                              },
                              Object {
                                "name": "Host 2",
                                "url": "#",
                              },
                              Object {
                                "name": "Host 3 with a very long name",
                                "url": "#",
                              },
                              Object {
                                "name": "Host 4",
                                "onClick": [Function],
                                "url": undefined,
                              },
                              Object {
                                "name": "Host 5",
                                "onClick": undefined,
                                "url": "#",
                              },
                            ]
              }
              totalPages={3}
      />,
            "_forcedUpdate": false,
            "_instance": [Circular],
            "_newState": null,
            "_rendered": <div
              className="breadcrumb-switcher"
              style={
                      Object {
                              "position": "relative",
                            }
              }
      >
              <BreadcrumbSwitcherToggler
                      onClick={[Function]}
              />
              <Overlay
                      animation={[Function]}
                      container={[Circular]}
                      onEnter={[Function]}
                      onHide={[Function]}
                      placement="bottom"
                      rootClose={true}
                      show={true}
                      target={[Function]}
              >
                      <BreadcrumbSwitcherPopover
                              currentPage={2}
                              hasError={false}
                              id="breadcrumb-switcher-popover"
                              loading={false}
                              onNextPageClick={[Function]}
                              onPrevPageClick={[Function]}
                              resources={
                                      Array [
                                              Object {
                                                "name": "Host 1",
                                                "url": "#",
                                              },
                                              Object {
                                                "name": "Host 2",
                                                "url": "#",
                                              },
                                              Object {
                                                "name": "Host 3 with a very long name",
                                                "url": "#",
                                              },
                                              Object {
                                                "name": "Host 4",
                                                "onClick": [Function],
                                                "url": undefined,
                                              },
                                              Object {
                                                "name": "Host 5",
                                                "onClick": undefined,
                                                "url": "#",
                                              },
                                            ]
                              }
                              totalPages={3}
                      />
              </Overlay>
      </div>,
            "_rendering": false,
            "_updater": [Circular],
          },
        },
      }
    }
    onEnter={[Function]}
    onHide={[Function]}
    placement="bottom"
    rootClose={true}
    show={true}
    target={[Function]}
  >
    <BreadcrumbSwitcherPopover
      currentPage={2}
      hasError={false}
      id="breadcrumb-switcher-popover"
      loading={false}
      onNextPageClick={[Function]}
      onPrevPageClick={[Function]}
      resources={
        Array [
          Object {
            "name": "Host 1",
            "url": "#",
          },
          Object {
            "name": "Host 2",
            "url": "#",
          },
          Object {
            "name": "Host 3 with a very long name",
            "url": "#",
          },
          Object {
            "name": "Host 4",
            "onClick": [Function],
            "url": undefined,
          },
          Object {
            "name": "Host 5",
            "onClick": undefined,
            "url": "#",
          },
        ]
      }
      totalPages={3}
    />
  </Overlay>
</div>
`;
